packages/shared/src/store/sync.ts

```diff
@@ -598,11 +596,24 @@ export const clearSyncQueue = () => {
   syncQueue.clear();
 };
 
-export const start = async () => {
-  api.subscribeUnreads(handleUnreadUpdate);
-  api.subscribeGroups(handleGroupUpdate);
-  api.subscribeToChannelsUpdates(handleChannelsUpdate);
-  api.subscribeToChatUpdates(handleChatUpdate);
-  api.subscribeToContactUpdates(handleContactUpdate);
-  useStorage.getState().start();
+export const initializeStorage = () => {
+  return syncQueue.add('initialize storage', async () => {
+    return useStorage.getState().start();
+  });
+};
+
+export const setupSubscriptions = async () => {
+  return syncQueue.add('setup subscriptions', async () => {
+    await Promise.all([
+      api.subscribeUnreads(handleUnreadUpdate),
+      api.subscribeGroups(handleGroupUpdate),
+    ]);
+    return Promise.all([
+      api.subscribeToChannelsUpdates(handleChannelsUpdate),
+      api.subscribeToChatUpdates(handleChatUpdate),
+      api.subscribeToContactUpdates(handleContactUpdate),
+    ]);
+  });
 };
+
+export const start = async () => {};
```

Comment:
if this is no op, should we remove?

Reply:
I've removed for now, and reworked init a little bit. We should really colocate all the start stuff, possibly in `useInitialSync`.